void_migration/initial.py

```diff
@@ -99,7 +107,7 @@ def set_concentration(s, X, Y, p):
     #     c[int(p.internal_geometry.perf_pts[0] * p.nx) : int(p.internal_geometry.perf_pts[1] * p.nx)] = 1
     #     c[int(p.internal_geometry.perf_pts[1] * p.nx) :] = 2
     #     c[np.isnan(s)] = np.nan
-    if p.charge_discharge:
+    if hasattr(p, "charge_discharge"):  # p.charge_discharge:
```

Comment:
we need to decide how we want to handle optional flags like this. my preference is for all flags to be listed in `defaults.json`, so that it is clear what their default value is. this should be `False` by default, and then the check `if p.discharge` would be clearer here.

Reply:
kept it as it is as I need to understand how to do it